bophono/PhonStateKVP.py

```diff
@@ -44,25 +42,27 @@ def doCombineCurEnd(self, endofword, nrc='', nextvowel=''): # nrc = next root co
         self.phon += self.end
 
 
-    def combineWithException(self, exception):
+    def combineWithException(self, exception, tibetanSyllable):
         syllables = exception.split('|')
         for syl in syllables:
             indexplusminus = syl.find('-')
             if indexplusminus == -1:
                 print("invalid exception syllable: "+syl)
                 continue
-            self.combineWith(syl[:indexplusminus], syl[indexplusminus+1:])
+            self.combineWith(syl[:indexplusminus], syl[indexplusminus+1:], tibetanSyllable)
 
-    def combineWith(self, nextroot, nextend):
+    def combineWith(self, nextroot, nextend, tibetanSyllable):
         nextrootconsonant = nextroot
         nextvowel = ''
         self.doCombineCurEnd(False, nextrootconsonant, nextvowel)
         self.position += 1
         if nextrootconsonant == "-":
             self.phon += ""
+        elif re.search(r'^བ[ོ]?[རསད]?(འི)?$', tibetanSyllable) and self.position == 1:
+            self.phon += "w"
         elif nextrootconsonant.startswith("dz") and self.position > 1:
             self.phon += "z"
-        elif nextrootconsonant.startswith("tr") and self.position == 1:
+        elif "གྲ" in tibetanSyllable and nextrootconsonant.startswith("tr") and self.position == 2:
```

Comment:
seems a bit hacky to me (and probably wrong but I'm not sure), please add comments

Reply:
I think this one should be removed too